### PR TITLE
rm_netxduo_ether receive path fixes regarding netx interface index and pointer

### PR DESCRIPTION
**Pull request by elrafoon**
Without this fixes, rm_netxduo_ether is only usable if it is the primary interface of given netxduo NX_IP stack.

But that's not always the case!

### CODE REVIEW

**Review by jritchey-renesas — CHANGES_REQUESTED** [review]
I agree, this is a bug in the existing driver.

Could you also apply this change to lines 788, and 833? The correct interface index needs to be provided to `_nx_ip_driver_link_status_event`.
```c
            /* Notify NetX that the link is up. */
            p_netxduo_ether_instance->p_ctrl->p_interface->nx_interface_link_up = NX_TRUE;
            _nx_ip_driver_link_status_event(p_netxduo_ether_instance->p_ctrl->p_ip,
                                            p_netxduo_ether_instance->p_ctrl->p_interface->nx_interface_index);
```
```c
            /* Notify NetX that the link is down. */
            p_netxduo_ether_instance->p_ctrl->p_interface->nx_interface_link_up = NX_FALSE;
            _nx_ip_driver_link_status_event(p_netxduo_ether_instance->p_ctrl->p_ip,
                                            p_netxduo_ether_instance->p_ctrl->p_interface->nx_interface_index);
```

File: ra/fsp/src/rm_netxduo_ether/rm_netxduo_ether.c [hunk 1]
```diff
@@ -332,7 +332,7 @@ void rm_netxduo_ether (NX_IP_DRIVER * driver_req_ptr, rm_netxduo_ether_instance_
         {
             /* Return the link status in the supplied return pointer.  */
             *(driver_req_ptr->nx_ip_driver_return_ptr) =
-                driver_req_ptr->nx_ip_driver_ptr->nx_ip_interface[p_ether_instance->p_cfg->channel].nx_interface_link_up;
+                driver_req_ptr->nx_ip_driver_ptr->nx_ip_interface[interface_ptr->nx_interface_index].nx_interface_link_up;
```
Comment (jritchey-renesas) [inline ra/fsp/src/rm_netxduo_ether/rm_netxduo_ether.c:335]:
```suggestion
                *(driver_req_ptr->nx_ip_driver_return_ptr) = interface_ptr->nx_interface_link_up;
```

**Comment by renesas-abigail** [comment]
This PR is tracked internally by FSPRA-3840.

**Comment by elrafoon** [comment]
Sorry for the delay, I was busy on a project.

Shall I do something, or is it resolved?

**Comment by jritchey-renesas** [comment]
Its no worries, the fix was added in the v5.6.0 release. Thanks for bringing the issue to our attention!

**Pull request closed without merging**